src/main/com/aca/homework/week8/car_repository/CarRepository.java

```diff
@@ -0,0 +1,33 @@
+package com.aca.homework.week8.car_repository;
+
+public class CarRepository {
+
+    private Car[] cars = new Car[1000];
+    private int pointer = 0;
+
+    public Car save(Car car) {
+        for (int i = 0; i < pointer; i++) {
+            if (cars[i].equals(car)) {
+                cars[i].update(car);
+                return cars[i];
+            }
+        }
+        cars[pointer] = car;
+        pointer++;
+        return cars[pointer -1];
+    }
+
+    public Car findByVin(String vin) {
+        if(pointer == 0)
+            return null;
+        for (Car car : cars) {
```

Comment:
here you will get a null pointer exception. also, there is no need to check the whole car array. the performance will be bad

Reply:
Why null pointer exception?